sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -872,7 +872,8 @@ class Analyzer(
           if (missingExpr.nonEmpty) {
             extractedExprBuffer += ne
           }
-          ne.toAttribute
+          // alias will be cleaned in the rule CleanupAliases
+          if (ne.isInstanceOf[Alias]) ne else ne.toAttribute
```

Comment:
Sure, let me change it to `ne` without calling `toAttribute` and move the test case to `DataFrameWindowSuite`. Thanks!